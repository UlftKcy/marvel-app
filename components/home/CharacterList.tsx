import { Character } from "@/types";
import Link from "next/link";
import {
    CharacterCard,
    CharacterCardHeader,
    CharacterCardImageWrapper,
  } from "@/components/ui/character/card";
import Image from "next/image";

export default function CharacterList(characters:Character[]) {
    return(
        Object.values(characters).map((character) => {
            const { id, name, thumbnail } = character as Character;
            return (
              <Link href={`/${id}`} key={id}>
                <CharacterCard>
                  <CharacterCardImageWrapper>
                    <Image
                      src={thumbnail.path + "." + thumbnail.extension}
                      width={100}
                      height={100}
                      objectFit="cover"
                      quality={100}
                      style={{
                        width: "100%",
                        height: "100%",
                        borderTopLeftRadius: "6px",
                        borderTopRightRadius: "6px",
                      }}
                      alt={name}
                    />
                  </CharacterCardImageWrapper>
                  <CharacterCardHeader>{name}</CharacterCardHeader>
                </CharacterCard>
              </Link>
            );
          })
    )
}