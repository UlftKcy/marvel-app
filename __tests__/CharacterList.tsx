import CharacterList from '@/components/home/CharacterList';
import { render } from '@testing-library/react';

const characters = [
    {
      id:1,
      name:"Name",
      description:"description",
      thumbnail:{
        path: "http://i.annihil.us/u/prod/marvel/i/mg/c/e0/535fecbbb9784",
        extension: "jpg"
      }
    }
  ]

it('renders CharacterList correctly', () => {
  const { asFragment } = render(<CharacterList {...characters} />);
  expect(asFragment()).toMatchSnapshot();
});
