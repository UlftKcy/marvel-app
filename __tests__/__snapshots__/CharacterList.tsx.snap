// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CharacterList correctly 1`] = `
<DocumentFragment>
  <a
    data-testid="1"
    href="/1"
  >
    <div
      class="sc-a7af3068-0 kuWxYq"
    >
      <div
        class="sc-a7af3068-1 mxHPK"
      >
        <img
          alt="Name"
          data-nimg="1"
          decoding="async"
          height="100"
          loading="lazy"
          src="/_next/image?url=http%3A%2F%2Fi.annihil.us%2Fu%2Fprod%2Fmarvel%2Fi%2Fmg%2Fc%2Fe0%2F535fecbbb9784.jpg&w=256&q=100"
          srcset="/_next/image?url=http%3A%2F%2Fi.annihil.us%2Fu%2Fprod%2Fmarvel%2Fi%2Fmg%2Fc%2Fe0%2F535fecbbb9784.jpg&w=128&q=100 1x, /_next/image?url=http%3A%2F%2Fi.annihil.us%2Fu%2Fprod%2Fmarvel%2Fi%2Fmg%2Fc%2Fe0%2F535fecbbb9784.jpg&w=256&q=100 2x"
          style="color: transparent; width: 100%; height: 100%; object-fit: cover; border-top-left-radius: 6px; border-top-right-radius: 6px;"
          width="100"
        />
      </div>
      <h4
        class="sc-a7af3068-2 iplHcB"
      >
        Name
      </h4>
    </div>
  </a>
</DocumentFragment>
`;
